src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_plugin.cpp

```diff
@@ -165,6 +166,27 @@ void AieTracePluginUnified::updateAIEDevice(void *handle, bool hw_context_flow)
   }
   AIEData.valid = true; // initialize struct
   
+  // Check if trace streams are available TODO
+  AIEData.metadata->setNumStreamsPLIO(
+      (db->getStaticInfo()).getNumAIETraceStream(deviceID, io_type::PLIO));
+  AIEData.metadata->setNumStreamsGMIO(
+      (db->getStaticInfo()).getNumAIETraceStream(deviceID, io_type::GMIO));
+  
+  uint64_t numStreamsPLIO = AIEData.metadata->getNumStreamsPLIO();
+  uint64_t numStreamsGMIO = AIEData.metadata->getNumStreamsGMIO();
+  bool isPLIO = (numStreamsPLIO > 0) ? true : false;
+  bool isGMIO = (numStreamsGMIO > 0) ? true : false;
+  
+  // Check if we've already configured a PLIO partition and current partition also has PLIO
+  // If so, skip this entire partition. GMIO-only partitions are still allowed. This is applicable just for register xclbin flow.
+  if ((db->getStaticInfo()).getAppStyle() == xdp::AppStyle::REGISTER_XCLBIN_STYLE && isPLIO && configuredOnePlioPartition) {
```

Comment:
We need to check here for "!isGMIO" as well.
For any aie partition with GMIO trace, following partitions with PLIO trace, "isPLIO" will be true as we check only the PL side. But partition with GMIO should still be configured for trace.